Editor/Window/UserProfileSelection.cs

```diff
@@ -0,0 +1,102 @@
+            {
+                BucketSelection(_bootstrapSettings.BucketName);
+                _bootstrapSettings.SaveSelectedBucket();
+            }
+        }
+
+        public void BucketSelection(string selectedBucket)
+        {
+            _bootstrapSettings.SelectBucket(selectedBucket);
+            _bootstrapSettings.SaveSelectedBucket();
+            _container.Q<Label>("S3BucketNameLabel").text = selectedBucket;
+        }
+
+        public void AccountSelection(bool isSetup)
+        {
+            _awsUserProfilesPage.RefreshProfiles();
+            var accountSelectFields = _container.Query<DropdownField>(null,"AccountSelection").ToList();
+            foreach (var accountSelect in accountSelectFields)
+            {
+                if (isSetup)
+                {
+                    accountSelect.RegisterValueChangedCallback(_ => { OnAccountSelect(accountSelect.index); });
+                }
+                
+                accountSelect.choices = _awsUserProfilesPage.UpdateModel.AllProlfileNames.ToList();
+                if (accountSelect.choices.Contains("default"))
+                {
+                    accountSelect.index = accountSelect.choices.IndexOf(_stateManager.SelectedProfile is "default" or null ? "default" : _stateManager.SelectedProfile);
+                }
+            }
+            
+        }
+
+        private void OnAccountSelect(int index)
+        {
+            var accountSelectLabels = _container.Query<Label>(null, "AccountSelectLabel").ToList();
+            BucketSelection();
+            foreach (var label in accountSelectLabels)
+            {
+                switch (label.name)
+                {
+                    case "S3BucketNameLabel":
+                        label.text = _bootstrapSettings.BucketName ?? "No Bucket Created";
+                        break;
+                    case "Region":
+                        if (_awsUserProfilesPage.UpdateModel.RegionBootstrap.RegionIndex >= 0)
+                        {
+                            label.text =
+                                _awsUserProfilesPage.UpdateModel.RegionBootstrap.AllRegions[
+                                    _awsUserProfilesPage.UpdateModel.RegionBootstrap.RegionIndex];
+                        }
+                        break;
+                    case "BootstrapStatus":
+                        label.text = _bootstrapSettings.BucketName != null ? "Active" : "Inactive";
+                        break;
+                }
+            }
+
+            UpdateModel(index);
+        }
+
+        private void UpdateModel(int index)
+        {
+            _awsUserProfilesPage.UpdateModel.SelectedProfileIndex = index;
+            _awsUserProfilesPage.UpdateModel.Update();
+            _stateManager.SelectedProfile = _awsUserProfilesPage.UpdateModel.AllProlfileNames[index];
+            _stateManager.CoreApi.PutSetting(SettingsKeys.CurrentProfileName,
+                _stateManager.SelectedProfile);
```

Comment:
StateManager can just have a SetSelectedProfile, which does both of this. I would assume you always want the two to occur together?